### ISSUE TITLE
AverageExpression calculations different after update to R 4.0.2

### ISSUE THREAD

**Issue by LotteSlenders** [satijalab/seurat#3616]
Recently I have installed R 4.0.2 upgrading after R 3.6.2. After installing I have noticed that the function AverageExpression does not return the same results before the updates. Both instances I worked with Seurat 3.2.2. 

```r
average.expression <- AverageExpression(object, 
                                        return.seurat = T, 
                                        #slot = "scale.data",
                                        features = rownames(GetAssayData(seuset, slot = "data")))

average.expression <- GetAssayData(average.expression, slot = "scale.data")


             | R 4.0.2      | R 3.6.2
A1CF         | -0.080903475 | -0.003663958
A2M          | -0.494818930 | -0.522476796
A2M-AS1     | 1.908577156  | 2.031097942
A4GALT      | -0.416873744 | -0.411218980


```

I don't know why or where the discrepancy arises, but the impact is significant.

**Comment by frederikziebell** [comment]
Is `object` exactly the same or are you doing some processing before that might be leading to a different `object`?

**Comment by LotteSlenders** [comment]
> Is `object` exactly the same or are you doing some processing before that might be leading to a different `object`?

Yes, the object is exactly the same. I load my object and calculate AverageExpresion directly after.

**Comment by chansigit** [comment]
That's scary!

**Comment by timoast** [comment]
I wasn't able to reproduce this. I ran the following code in R 3.6.2 and R 4.0.1 and got identical results (shown below):

```r
library(Seurat)
av.exp <- AverageExpression(pbmc_small, return.seurat = TRUE)
head(GetAssayData(av.exp, slot = "scale.data"))
```

```
                  0           1         2
MS4A1    -0.7726395 -0.35682968 1.1294692
CD79B    -0.7343496 -0.40454281 1.1388924
CD79A    -0.9805369 -0.03785132 1.0183882
HLA-DRA  -1.1018052  0.25170530 0.8500999
TCL1A    -0.8456328 -0.25812323 1.1037561
HLA-DQB1 -1.0755121  0.17381206 0.9017001
```

**Comment by josegarciamanteiga** [comment]
Same for me. I think I got the reason why. Not sure how to use 4.0 to obtain the same results as with 3.
In 4.0 there is a new option for the function, 

group.by | Categories for grouping (e.g, ident, replicate, celltype); 'ident' by default

While in the 3.2 version was not.
I'm trying to force the 4.0 version not to group by anything. Tried with 'none' NULL or other but without success. I guess it is useful as datasets get more complicated with replicates and factorial designs, but it does not allow the old behaviour.
-- | --
